common/persistence/versionhistory/version_history.go

```diff
@@ -33,107 +33,63 @@ import (
 	"go.temporal.io/server/common"
 )
 
-// NewItem create a new version history item
-func NewItem(
-	inputEventID int64,
-	inputVersion int64,
-) *historyspb.VersionHistoryItem {
-
-	if inputEventID < 0 || (inputVersion < 0 && inputVersion != common.EmptyVersion) {
-		panic(fmt.Sprintf(
-			"invalid version history item event ID: %v, version: %v",
-			inputEventID,
-			inputVersion,
-		))
+// NewVersionHistory create a new instance of VersionHistory.
+func NewVersionHistory(branchToken []byte, items []*historyspb.VersionHistoryItem) *historyspb.VersionHistory {
+	return &historyspb.VersionHistory{
+		BranchToken: branchToken,
+		Items:       items,
 	}
-
-	return &historyspb.VersionHistoryItem{EventId: inputEventID, Version: inputVersion}
-}
-
-// DuplicateItem duplicate VersionHistoryItem
-func DuplicateItem(item *historyspb.VersionHistoryItem) *historyspb.VersionHistoryItem {
-	return NewItem(item.EventId, item.Version)
 }
 
-// New create a new version history
-func New(
-	inputToken []byte,
-	inputItems []*historyspb.VersionHistoryItem,
-) *historyspb.VersionHistory {
-
-	token := make([]byte, len(inputToken))
-	copy(token, inputToken)
-	versionHistory := &historyspb.VersionHistory{
-		BranchToken: token,
-		Items:       nil,
-	}
+// CopyVersionHistory copies VersionHistory.
+func CopyVersionHistory(v *historyspb.VersionHistory) *historyspb.VersionHistory {
+	token := make([]byte, len(v.BranchToken))
+	copy(token, v.BranchToken)
 
-	for _, item := range inputItems {
-		if err := AddOrUpdateItem(versionHistory, DuplicateItem(item)); err != nil {
-			panic(fmt.Sprintf("unable to initialize version history: %v", err))
-		}
+	var items []*historyspb.VersionHistoryItem
+	for _, item := range v.Items {
+		itemCopy := *item
+		items = append(items, &itemCopy)
```

Comment:
is this making a copy?

Reply:
Yes: https://play.golang.org/p/RMsCFJokeMw

Reply:
```go
func main() {
	item1 := &Item{id: "id2"}
	item2 := &*item1
	item1.id = "id1"
	fmt.Println(item1,item2)
}
```

this prints `&{id1} &{id1}`
if you want to keep this change, can you add an comment here making sure `&*` is not used?

Reply:
to me this will be more error prone

Reply:
Who could write `&*`? Funny, but we have few occurrences. Lucky all in tests. I would expect `&` operator not to work here at all. The same as `&1` or `&"qwe"` don't work. But apparently here, compiler just collapse `&` in `*` and produce noop. And this totally make sense for me. I wouldn't expect this code to copy struct content.

Reply:
> Funny, but we had few occurrences.

For me, I would not rely on something error prone. But since you are making the change, up to you